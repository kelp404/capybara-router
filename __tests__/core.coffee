history = require 'history'
core = require '../lib/core'


test 'Get the current route.', ->
  core.setup
    history: history.createMemoryHistory
      initialEntries: ['/']
    routes: [
      name: 'web'
      uri: '/'
    ]
  route = core.getCurrentRoute()
  expect(route).toMatchSnapshot()

test 'Find the route by the name.', ->
  core.setup
    history: history.createMemoryHistory
      initialEntries: ['/']
    routes: [
      name: 'web'
      uri: '/'
    ]
  route = core.findRouteByName 'web', core.routes
  expect(route).toMatchSnapshot()

test 'Find the route by the location.', ->
  core.setup
    history: history.createMemoryHistory
      initialEntries: ['/users/AWgrmJp1SjjuUM2bzZXM/projects?index=0&sort=asc']
    routes: [
      name: 'web'
      uri: '/users/{userId:[\\w-]{20}}/projects?index?sort'
    ]
  route = core.findRoute core.history.location
  expect(route).toMatchSnapshot()
