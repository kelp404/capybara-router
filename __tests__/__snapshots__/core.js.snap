// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find the route by the location. 1`] = `
Route {
  "component": undefined,
  "isAbstract": false,
  "matchPattern": "/users/([\\\\w-]{20})/projects",
  "matchReg": /\\^\\\\/users\\\\/\\(\\[\\\\w-\\]\\{20\\}\\)\\\\/projects\\$/,
  "name": "web",
  "onEnter": undefined,
  "parents": Array [],
  "resolve": Object {},
  "uri": "/users/{userId:[\\\\w-]{20}}/projects?index?sort",
  "uriParamKeys": Array [
    "userId",
    "?index",
    "?sort",
  ],
  "uriTemplate": "/users/{userId}/projects",
}
`;

exports[`Find the route by the name. 1`] = `
Route {
  "component": undefined,
  "isAbstract": false,
  "matchPattern": "/",
  "matchReg": /\\^\\\\/\\$/,
  "name": "web",
  "onEnter": undefined,
  "parents": Array [],
  "resolve": Object {},
  "uri": "/",
  "uriParamKeys": Array [],
  "uriTemplate": "/",
}
`;

exports[`Get the current route. 1`] = `
Route {
  "component": undefined,
  "isAbstract": false,
  "matchPattern": "/",
  "matchReg": /\\^\\\\/\\$/,
  "name": "web",
  "onEnter": undefined,
  "parents": Array [],
  "resolve": Object {},
  "uri": "/",
  "uriParamKeys": Array [],
  "uriTemplate": "/",
}
`;
